README.md

```diff
@@ -94,44 +94,64 @@ By default, the `fal run` command runs the Python scripts as a post-hook, **only
+`fal` integrates with `dbt`'s `profile.yml` file to access and read data from the data warehouse. Once you setup credentials in your `profile.yml` file for your existing `dbt` workflows anytime you use `ref` or `source` to create a dataframe `fal` authenticates using the credentials specified in the `profile.yml` file.
 
 ## `meta` Syntax
 ```yaml
 models:
   - name: historical_ozone_levels
     ...
     meta:
+      owner: "@me"
       fal:
         scripts:
           - send_slack_message.py
-          - another_python_script.py # will be ran sequentially
+          - another_python_script.py # will be run sequentially
 ```
 Use the `fal` and `scripts` keys underneath the `meta` config to let `fal` CLI know where to look for the Python scripts. You can pass a list of scripts as shown above to run one or more scripts as a post-hook operation after a `dbt run`.
 
-## Variables
-Inside a Python script, you get access to some useful variables:
+## Variables and functions
+Inside a Python script, you get access to some useful variables and functions:
+
+A `context` object with information relevant to the model through which the script was run. For the [`meta` Syntax](#meta-syntax) example, we would get the following:
 
 ```python
-# send_slack_message.py
+context.current_model.name
+#= historical_ozone_levels
+
+context.current_model.meta
+#= {'owner': '@me'}
+
+context.current_model.meta.get("owner")
+#= '@me'
+
+context.current_model.status
+# Could be one of
+#= 'success'
+#= 'error'
+#= 'skipped'
+```
 
-# Refer to dbt models or sources by name. Returns a pandas.DataFrame.
+`fal` also includes some familiar functions from dbt as well as adding a couple of new useful ones
+
+```python
+# Refer to dbt models or sources by name and returns it as `pandas.DataFrame`
 ref('model_name')
 source('source_name', 'table_name')
 
-# Refer to a fal Context, which provides useful attributes about the dbt model in context.
-context.current_model.name # Returns the current model name
-
-context.current_model.meta # Returns the meta config of the current model as a Python dict
-context.current_model.meta["owner"] # will return the "owner" key in the meta config
+# You can use it to get the running model data
+ref(context.current_model.name)
+```
 
-context.current_model.status # Returns the status of the last dbt run
+It is also possible to send data back to your datawarehouse. This makes it easy to get the data, process it and upload it back into dbt territory.
 
-ref(context.current_model.name) # Returns a pandas.DataFrame for the current model in context
+```python
+# Upload a `pandas.DataFrame` back to the datawarehouse
+write_to_source(df, 'source_name', 'table_name2')
```

Comment:
lets put them under a sub header.. 

## Variables and functions 

### ref function 

### write_to_source function